### ISSUE TITLE
Choose License

### ISSUE THREAD

**Issue by ODXT** [devRant-Community/Community-Programming-Book#17]
COMMENT - @FilipeRamalho
[I think this decision should be met by ODXT as he begun the Project and else there would be a fight over which license.]

**Comment by ODXT** [comment]
I'm working on this with Michelle...

**Comment by gadilashashank** [comment]
My opinion:
I suggest to look at the way academic papers are licensed... usually free to view(the paid ones are charged by the publishers and not the authors), copyrighted (if all agree we could copyright the parts not picked up from other sources but then ALL sources referred must be strictly acknowledged), and can be referred to  by citing(kind of) the original project(the current repo). These points need to be discussed with all participants. Regarding distribution @ODXT as you are the owner you need to decide if you would want this work to be published as a proper printed book, or if you want to maintain a website and distribute electronic copies of the book.

**Comment by RootPixl** [comment]
@gadilashashank in my opinion that's good suggestion
Also, two versions of a book, printed and e-book are good options, you know, some like e-books, some prefer printed ones :blush:
Other thing, we could also try electronic signatures of authors in printed edition or kinda something
On the other hand, some people they can object that the electronic version would be free, and the print would be payable, so good option is naming printed one 'Fan Edition' and then adding few stickers with that version of book, everyone love stickers.

**Comment by FilipeRamalho** [comment]
I just want to note down, that whe are far away from print. It should only considered when we have a first version.

**Comment by RootPixl** [comment]
Maybe our own license? 
Like @gadilashashank said? 
And back to printed version, I know, just wondering 'bout it

**Comment by FilipeRamalho** [comment]
@RootPixl Had also that thought, especially considering a print version. The license could limit printing rights to ODXT.

**Comment by RootPixl** [comment]
Yeah, it could, it was his idea, and that's the good thing to limit printing rights only to ODXT

**Comment by gadilashashank** [comment]
I think the primary decision with the issue said lies mostly with @ODXT and the discussion part @FilipeRamalho , would possibly help in making a decison.

**Comment by FilipeRamalho** [comment]
Maybe like this ?

**Comment by gadilashashank** [comment]
@ODXT do you think this should be closed now that we have a license in the repo? Maybe we can re-open it later?

**Comment by KrzysztofSzewczyk** [comment]
We don't have license.

**Comment by Unrooted** [comment]
# Writeup
I actually think we shall write our own license. We're currently operating on a Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License and if we're looking forward to print our book when we will achieve 25-30 chapters, we shall write our own license to have a right to make money from the printed version. Just thinking in a bigger format, thinking more about the future. For now, our current license is OK, but we need to keep that in mind, if one day, we'd really like to print this, it's gonna be changed.

**Comment by MasterOfTheTiger** [comment]
Don't you have to get permission from all contributors to change a license to one which is incompatable?

**Comment by KrzysztofSzewczyk** [comment]
It would be better to start it right now

**Comment by Unrooted** [comment]
[![](https://api.gh-polls.com/poll/01D48T4D112NPTS7F54E4FY7P9/writing%20our%20own%20license)](https://api.gh-polls.com/poll/01D48T4D112NPTS7F54E4FY7P9/writing%20our%20own%20license/vote)
[![](https://api.gh-polls.com/poll/01D48T4D112NPTS7F54E4FY7P9/staying%20with%20the%20current%20one%20(and%20not%20printing%20the%20book))](https://api.gh-polls.com/poll/01D48T4D112NPTS7F54E4FY7P9/staying%20with%20the%20current%20one%20(and%20not%20printing%20the%20book)/vote)
[![](https://api.gh-polls.com/poll/01D48T4D112NPTS7F54E4FY7P9/looking%20to%20find%20something%20suitable%20for%20us%20(can%20be%20hard))](https://api.gh-polls.com/poll/01D48T4D112NPTS7F54E4FY7P9/looking%20to%20find%20something%20suitable%20for%20us%20(can%20be%20hard)/vote)

**Comment by FilipeRamalho** [comment]
If the License is to have restricitions I would advise to use an existing one. If you want to write an license you certainly need legal advice.

**Comment by KrzysztofSzewczyk** [comment]
[RELICENSING](https://github.com/devRant-Community/Community-Programming-Book/blob/master/RELICENSING) has been around for a long time. We should really revive this topic.
